client/features/output-visualization-page/components/UndoAndRedo.tsx

```diff
@@ -16,38 +17,26 @@ const useTemporalStore = <T,>(
 
 const UndoAndRedo = () => {
   const { undo, redo } = useTemporalStore((state) => state);
+  const canUndo = useTemporalStore((state) => state.pastStates.length > 0);
+  const canRedo = useTemporalStore((state) => state.futureStates.length > 0);
 
   return (
     <>
       <ButtonGroup orientation='vertical'>
         <IconButton
+          disabled={!canUndo}
           aria-label='undo'
           size='small'
-          sx={{
-            color: 'black',
-            backgroundColor: '#FFF',
-            '&:hover': { backgroundColor: '#f4f4f4' },
-            border: '0.5px solid #f4f4f4',
-            borderRadius: '0px',
-            width: '26px',
-            height: '26px',
-          }}
+          className={styles.iconButton}
```

Comment:
Instead of `iconButton`, something like `undoredo_button` is more readable and better adheres to naming best practices.